engine/kv_engine.cpp

```diff
@@ -2032,25 +2137,32 @@ Status KVEngine::listRegisterRecovered() {
 }
 
 template <typename DelayFree>
-Status KVEngine::listDestroy(List* list, DelayFree delay_free) {
+Status KVEngine::listDestroy(List* list, DelayFree delay_free, bool local) {
   // Currently, every list operation locks the whole list
   // and delay_free is not necessary.
   // We use delay_free here so that we can enable some lockless
   // operations in the future.
   while (list->Size() > 0) {
-    auto token = version_controller_.GetLocalSnapshotHolder();
-    list->PopFront(
-        [&](DLRecord* elem) { delay_free(elem, token.Timestamp()); });
+    auto ts = local ? version_controller_.GetLocalSnapshotHolder().Timestamp()
```

Comment:
This is invalid use of a RAII object.
This token should outlive the list->PopFront() function.
Actually destroy a List does not need the token, you can use version_controller.GetTimeStamp();